packages/sdk38/src/lcdapi/staking.spec.ts

```diff
@@ -0,0 +1,508 @@
+      const response = await client.staking.delegatorValidator(faucet.address, validatorAddress);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: {
+          operator_address: "cosmosvaloper1gjvanqxc774u6ed9thj4gpn9gj5zus5u32enqn",
+          consensus_pubkey:
+            "cosmosvalconspub1zcjduepqau36ht2r742jh230pxlu4wjmwcmkwpeqava80acphsu87vt5xlpqx6g7qh",
+          jailed: false,
+          status: 2,
+          tokens: jasmine.stringMatching(nonNegativeIntegerMatcher),
+          delegator_shares: jasmine.stringMatching(bigDecimalMatcher),
+          description: {
+            moniker: "testing",
+            identity: "",
+            website: "",
+            security_contact: "",
+            details: "",
+          },
+          unbonding_height: "0",
+          unbonding_time: "1970-01-01T00:00:00Z",
+          commission: {
+            commission_rates: {
+              rate: "0.100000000000000000",
+              max_rate: "0.200000000000000000",
+              max_change_rate: "0.010000000000000000",
+            },
+            update_time: "2020-06-03T06:01:17.4747987Z",
+          },
+          min_self_delegation: "1",
+        },
+      });
+    });
+  });
+
+  describe("delegation", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegation(faucet.address, validatorAddress);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: {
+          delegator_address: faucet.address,
+          validator_address: validatorAddress,
+          shares: jasmine.stringMatching(bigDecimalMatcher),
+          balance: { denom: "ustake", amount: jasmine.stringMatching(nonNegativeIntegerMatcher) },
+        },
+      });
+    });
+  });
+
+  describe("unbondingDelegation", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.unbondingDelegation(faucet.address, validatorAddress);
+      expect(response.height).toMatch(nonNegativeIntegerMatcher);
+      expect(response.result!.delegator_address).toEqual("cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6");
+      expect(response.result!.validator_address).toEqual(
+        "cosmosvaloper1gjvanqxc774u6ed9thj4gpn9gj5zus5u32enqn",
+      );
+      expect(response.result!.entries.length).toBeGreaterThanOrEqual(1);
+      expect(response.result!.entries[0]).toEqual({
+        creation_height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        completion_time: jasmine.stringMatching(dateTimeStampMatcher),
+        initial_balance: "100",
+        balance: "100",
+      });
+    });
+  });
+
+  describe("redelegations", () => {
+    it("works", async () => {
+      // TODO: Set up a result for this test
```

Comment:
👍 No idea what a redelegation is …

Reply:
https://docs.cosmos.network/master/modules/staking/02_state_transitions.html#begin-redelegation

Basically an unbond plus a delegation in one step.